crates/bevy_ecs/src/reflect/mod.rs

```diff
@@ -40,5 +40,3 @@ impl DerefMut for AppTypeRegistry {
         &mut self.0
     }
 }
-
-impl_reflect_value!((in bevy_ecs) Entity(Hash, PartialEq, Serialize, Deserialize));
```

Comment:
I guess this macro was here because this module is `cfg`-gated behind the `bevy_reflect` feature, thus didn't suffer from this problem. So this is one way to solve the build error, the other being the use of `#[cfg_attr(...)]` to apply the `Reflect` derive macro (and its connected attributes) only when the `bevy_reflect` feature is enabled.